SCRAM/utils/backend-setup.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash -e
+backend=$(echo $1 | cut -d- -f2 | cut -d: -f1)
+req_type=$(echo $1 | cut -d- -f1)
+CAPS=$(echo $1 | cut -d- -f2 | cut -d: -f2)
+TOOL=config/toolbox/${SCRAM_ARCH}/tools/selected/${backend}.xml
+
+if [ ! -e ${TOOL} ] ; then
+  echo "Error: Unknown backend ${backend}" >&2
+  exit 1
+fi
+
+if [ "${CAPS}" = "reset" ] ; then
+  cp -f $CMSSW_RELEASE_BASE/${TOOL} ${TOOL}.tmp
+elif [ "${backend}" = "cuda" ] ; then
+  if [ "${CAPS}" = "native" ] ; then
+    DOTS=$(cudaComputeCapabilities | awk '{ print $2 }' | sort -u)
+    CAPS=$(echo $DOTS | sed -e 's#\.*##g')
+    if [ "${CAPS}" = "" ] ; then
+      echo "Warning: Unable to find cuda compute capabilities." >&2
+      exit 0
+    fi
+  else
+    CAPS=$(echo ${CAPS} | tr ',' '\n' | sed 's|^sm_||' | tr '\n' ' ' | sed 's| *$||')
+  fi
+  cp -f ${TOOL} ${TOOL}.tmp
+  # remove existing capabilities
+  sed -i -e "s# *-gencode arch=compute_..,code=sm_.. *# #g" ${TOOL}.tmp
+  sed -i -e "s# *-gencode arch=compute_..,code=\[sm_..,compute_..\] *# #g" ${TOOL}.tmp
+
+  # add support for the capabilities found on this machine
+  for CAP in $CAPS; do
+    sed -i -e "/flags CUDA_FLAGS=/s#=\"#=\"-gencode arch=compute_$CAP,code=[sm_$CAP,compute_$CAP] #" ${TOOL}.tmp
+  done
+  CAPS=$(echo $CAPS | tr ' ' '\n' | sed 's|^|sm_|' | tr '\n' ',' | sed 's|,$||')
+elif [ "${backend}" = "rocm" ] ; then
+  if [ "${CAPS}" = "native" ] ; then
+    CAPS=$(rocmComputeCapabilities | awk '{ print $2 }' | sort -u)
+    if [ "${CAPS}" = "" ] ; then
+      echo "Warning: Unable to find rocm compute capabilities." >&2
+      exit 0
+    fi
+  else
+    CAPS=$(echo ${CAPS} | tr ',' ' ')
+  fi
+  cp -f ${TOOL} ${TOOL}.tmp
+  # remove existing capabilities flags
+  sed -r -i -e '/ROCM_FLAGS/s/--offload-arch=gfx[0-9a-f]+//g' ${TOOL}.tmp
+  # remove empty ROCM_FLAGS lines
+  sed -r -i -e '/ROCM_FLAGS=" *"/d' ${TOOL}.tmp
+
+  #add support for the capabilities found on this machine
+  for CAP in $CAPS; do
+    sed -i -e "s#</client>#</client>\n  <flags ROCM_FLAGS=\"--offload-arch=${CAP}\"/>#" ${TOOL}.tmp
+  done
```

Comment:
@fwyzard , may be something like this should be better for cuda tool too

Reply:
But if you mean adding it as a separate entry only when done by this script, sure, OK for me.

Reply:
yes, separate entry when done by the script. Basically it will remove all existing `-gencode arch=compute_75,code=[sm_75,compute_75]` entries from the tool file , delete any empty CUDA_FLAGS line and then add new entries on new like after `</client>`. This does not require changing existing cuda tool file and should also work if we decided to change the cuda file to have multiple `CUDA_FLAGS` lines